### PR TITLE
ExtJS 7

### LINKED ISSUE #6362: ExtJS 7

### PR DESCRIPTION
**Pull request by weisswurstkanone**
resolves #6362

### CODE REVIEW

**Comment by dpfaffenbauer** [comment]
:clap: :open_mouth:

File: bundles/AdminBundle/Controller/Admin/MiscController.php [hunk 1]
```diff
@@ -778,4 +784,780 @@ public function testAction(Request $request)
...
+     * @throws \Exception
+     */
+    public function pimcoreInternalScriptsAction(Request $request) {
+
+        $bundleManager = $this->get('pimcore.extension.bundle_manager');
+
+        $pluginJsPaths = $bundleManager->getJsPaths();
+        $locale = \Pimcore::getContainer()->get('pimcore.locale')->getLocale();
+
+        $scripts = [
```
Comment (dvesh3) [inline bundles/AdminBundle/Controller/Admin/MiscController.php:801]: can we move this to a config file? and read array

File: bundles/AdminBundle/Controller/Admin/MiscController.php [hunk 2]
```diff
@@ -778,4 +784,780 @@ public function testAction(Request $request)
...
+     *
+     * @throws \Exception
+     */
+    public function pimcoreEditmodeScriptsAction(Request $request, Packages $packages, RouterInterface $router) {
+        $scriptContents = "";
+
+        $bundleManager = $this->get('pimcore.extension.bundle_manager');
+        $pluginJsPaths = $bundleManager->getEditmodeJsPaths();
+
+        $scripts = [
```
Comment (dvesh3) [inline bundles/AdminBundle/Controller/Admin/MiscController.php:1492]: move to config and read array

File: bundles/AdminBundle/Resources/public/css/ext-modifications.css [hunk 3]
```diff
@@ -646,3 +654,182 @@ body > .x-mask {
...
+.x-window-header-title-default {
+    color: #F0F0F0
+}
+
+.x-toolbar-text-default {
+    color: #4b4c4c;
+    font: 300 13px/16px 'Open Sans', 'Helvetica Neue', helvetica, arial, verdana, sans-serif;
+}
+
+/*.x-tab-bar-default {*/
```
Comment (dvesh3) [inline bundles/AdminBundle/Resources/public/css/ext-modifications.css:688]: remove please

File: bundles/AdminBundle/Resources/public/css/ext-modifications.css [hunk 4]
```diff
@@ -646,3 +654,182 @@ body > .x-mask {
...
+#pimcore_panel_tabs > .x-tab-bar {
+    background: #0C0F12;
+    background-image: url(/bundles/pimcoreadmin/img/logo-gray.svg);
+    background-repeat: no-repeat;
+    background-position: right 5px top 4px;
+    background-size: 105px;
+}
+
+
+/*.x-tab-inner-default {*/
```
Comment (dvesh3) [inline bundles/AdminBundle/Resources/public/css/ext-modifications.css:704]: remove please

File: bundles/AdminBundle/Controller/Admin/MiscController.php [hunk 5]
```diff
@@ -778,4 +784,780 @@ public function testAction(Request $request)
...
+
+
+    /**
+     * @Route("/pimcoreInternalScripts", name="pimcore_admin_pimcore_internal_scripts", methods={"GET"})
+     *
+     * @throws \Exception
+     */
+    public function pimcoreInternalScriptsAction(Request $request) {
+
+        $bundleManager = $this->get('pimcore.extension.bundle_manager');
```
Comment (dvesh3) [inline bundles/AdminBundle/Controller/Admin/MiscController.php:796]: please use DI

File: bundles/AdminBundle/Controller/Admin/MiscController.php [hunk 6]
```diff
@@ -778,4 +784,780 @@ public function testAction(Request $request)
     {
         return new Response('done');
     }
+
+
+    /**
+     * @Route("/pimcoreInternalScripts", name="pimcore_admin_pimcore_internal_scripts", methods={"GET"})
```
Comment (dvesh3) [inline bundles/AdminBundle/Controller/Admin/MiscController.php:790]: pimcore-internal-scripts

File: bundles/AdminBundle/Controller/Admin/MiscController.php [hunk 7]
```diff
@@ -778,4 +784,780 @@ public function testAction(Request $request)
...
+        }
+
+
+
+    }
+
+
+
+    /**
+     * @Route("/pimcoreEditmodeScripts", name="pimcore_admin_pimcore_editmode_scripts", methods={"GET"})
```
Comment (dvesh3) [inline bundles/AdminBundle/Controller/Admin/MiscController.php:1482]: pimcore-editmode-scripts

**Comment by brusch** [comment]
Todo: 

- how-to guide how to upgrade Ext version (step-by-step guide)
- I'd put all Ext related stuff into a folder: 
  - `public/css/ext-js/` -> `PimcoreApp-all_*.css` ... 
  - `public/js/ext-js/` -> `pimcore*.json` files, `bootstrap.js`, ...
- `appEditmode.js` should go to `document/edit/` where the other editmode related stuff lives 
- do we really need all themes? Not really, right? Currently there is `neptune`, `material` and `triton` we should only have in Git what's acutally needed.

**Comment by brusch** [comment]
It also seems that some modules are checked in multiple times ... that needs to be checked as well: 
e.g.: https://github.com/pimcore/pimcore/tree/216d1fb78a68d22207b639e3497f9341b4ae25d5/bundles/AdminBundle/Resources/public/js/lib/node_modules/@sencha/ext-classic-theme-material/node_modules/@sencha

**Comment by weisswurstkanone** [comment]
closed in favour of #7824

**Pull request closed without merging**